server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -231,6 +263,23 @@ public int hashCode() {
         return hashCode;
     }
 
+    /**
+     * Checks if 2 instances are equal, ignoring optionalRepos for both.
+     *
+     * @param o other instance
+     * @return {@code true} iff both instances are equal, not including the repositories in both instances if they are part of optionalRepos.
+     */
+    public boolean equalsIgnoreOptionalRepo(Object o) {
```

Comment:
Why not override `equals` itself? If we compare RemoteStoreNodeAttribute in some other place, unless we have context, we will use `equals` (or some `Collections` methods will use it implicitly)

Reply:
This is a specific usecase while node joins, didn't want to change any other behavior for the class.